project/Hooje/README.md

```diff
@@ -0,0 +1,59 @@
+Simple Parallel Decision Tree in C++
+====
+
+# Basic Information
+Traditional Machine Learning models are still effective ways for classification.
+And Decision Tree is one of them.
+
+Decision Tree is a series a node that start with one root node and extends to many nodes that can represents the classfication rules for the data.
+For a dataset, Decision Tree will calculate the impurity (GINI index, entropy...) of every possible split, and choose the best one to be the rule of root node and create two leaf nodes.
+For each node, use the same method to decide the rule and create its child nodes.
+
+
+# Problem to Solve
+Decision Tree is an effective model in many classification senerio. However, it takes so much time to run because it cannot be speedup by CUDA like Neural Network, especially when there are a lot of features in that Dataset.
+
+Thus, I want to implement a simple version of Decision Tree in C++ (can be used from Python) from scratch to reduce the execution time. Besides, I would like to modify some of its functions so that it can be executed parallelly.
+
+# Prospective Users
+Those people who use traditional ML models can benefit from my Decision Tree. 
+They can spent less time waiting for the model to finish.

```

Comment:
Do you plan to make the faster runtime a goal of the project?

Reply:
Yes, I assume that Decision Tree in C++ can run faster than which in python.
And I firmly believe that it can speed up with SIMD or multi-thread.

Reply:
It will be a good practice to provide both Python and C++ API in your library.  Engineering-wise, it is more valuable than parallelism.  You may consider to add that to the proposal and we can see how the plan looks like.

Reply:
I want to ask what kind of API will be valuable?
Do you mean those basic ones ? (those provided in Python Library)
Or I need to create a new one related to this project? Like, user can choose to use SIMD.

Reply:
It is usually hard if not impossible to tell what API is important before making them.  Guesses should be made in a proposal and let the code prove.